iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.List;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.checks.utils.CheckUtils;
+import org.sonar.iac.docker.checks.utils.CommandDetector;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S6437")
+public class SecretsGenerationCheck implements IacCheck {
+
+  private static final String MESSAGE = "Revoke and change this secret, as it might be compromised.";
+
+  private static final Set<String> SENSITIVE_KEYTOOL_FLAGS = Set.of("-gencert", "-genkeypair", "-genseckey", "-genkey");
+
+  private static final CommandDetector KEYTOOL_DETECTOR = CommandDetector.builder()
+    .with("keytool")
+    .withOptionalRepeatingExcept(SENSITIVE_KEYTOOL_FLAGS)
+    .with(SENSITIVE_KEYTOOL_FLAGS)
+    .withOptionalRepeatingExcept(SENSITIVE_KEYTOOL_FLAGS)
+    .build();
```

Comment:
Same things that I mentioned on @mstachniuk PR, here we are potentially matching over several commands, which in some case may throw strange issue or even FP.
```
RUN keytool -random1 && unrelated_command -genkeypair 
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
This is indeed an edgy case here, but I don't exclude that something like that can happen.
We can keep it as it is for now and see if we get FP report later from users, but we need to keep in mind such thing as we usually like conservative approach.
Please add this use case in test file to document it.

Still without a real bash parser, we cannot do much better.

Reply:
I will add a FP, but i think the new CommandDetector can handle this aswell